lnwallet/btcwallet/signer.go

```diff
@@ -73,6 +76,94 @@ func deriveFromKeyLoc(scopedMgr *waddrmgr.ScopedKeyManager,
 	return addr.(waddrmgr.ManagedPubKeyAddress).PrivKey()
 }
 
+// deriveKeyByBIP32Path derives a key described by a BIP32 path. The path is
+// expected to have 2^31 added to a hardened key in the path.
+func (b *BtcWallet) deriveKeyByBIP32Path(path []uint32) (*btcec.PrivateKey,
```

Comment:
Curious about the relationship of `btcsuite/btcwallet` and `lnwallet/btcwallet`. Should the HD wallet related implementations live in `btcsuite` since it's quite low level?

Reply:
The `btcsuite` one is agnostic of any of our key scopes. So the fact that we have an internal (`1017'`) and two external (`49'` and `84'`) scopes is not known to it. Therefore the logic in the `deriveKeyByBIP32Path()` that switches between the internal and external scopes must be in `lnd`.